PROTO_tests/tests/symbol_table_test.py

```diff
@@ -0,0 +1,766 @@
+        assert s.registered_name == "MyStrings"
+        assert c.registered_name == "MyCat"
+
+        reg = ak.list_registry()
+        assert a.registered_name in reg["Objects"]
+        assert s.registered_name in reg["Objects"]
+        assert c.registered_name in reg["Objects"]
+
+        # validate that all objects are attached
+        att = ak.attach_all(["MyArray", "MyCat"])
+        assert "MyArray" in att
+        assert "MyCat" in att
+        assert isinstance(att["MyArray"], ak.pdarray)
+        assert isinstance(att["MyCat"], ak.Categorical)
+        assert att["MyArray"].to_list() == a.to_list()
+        assert att["MyCat"].to_list() == c.to_list()
+
+        # validate all objects are unregistered
+        ak.unregister_all(["MyStrings", "MyArray"])
+        reg = ak.list_registry()
+        assert "MyArray" not in reg["Objects"]
+        assert "MyStrings" not in reg["Objects"]
+        assert "MyCat" in reg["Objects"]
+
+    @pytest.mark.parametrize("dtype", DTYPES)
+    def test_attach_weak_binding(self, dtype):
+        """
+        Ultimately pdarrayclass issues delete calls to the server when a bound object goes out of scope,
+        if you bind to a server object more than once and one of those goes out of scope it affects
+        all other references to it.
+        """
+        a = self.make_pdarray(dtype, 10).register("a_reg")
+        assert str(a)
+        b = ak.attach("a_reg")
+        b.unregister()
+        b = None  # Force out of scope
+        with pytest.raises(RuntimeError):
+            str(a)
+
+
+    def test_symentry_cleanup(self):
+        pda = self.make_pdarray(ak.int64, 10)
+        assert len(ak.list_symbol_table()) > 0
+        pda = None
+        assert len(ak.list_symbol_table()) == 0
+
+        s = self.make_pdarray(ak.str_, 10)
+        assert len(ak.list_symbol_table()) > 0
+        s = None
+        assert len(ak.list_symbol_table()) == 0
+
+        cat = ak.Categorical(self.make_pdarray(ak.str_, 10))
+        assert len(ak.list_symbol_table()) > 0
+        cat = None
+        assert len(ak.list_symbol_table()) == 0
+
+        seg = ak.SegArray(
+            ak.arange(0, 10, 2), self.make_pdarray(ak.float64, 10)
+        )
+        assert len(ak.list_symbol_table()) > 0
+        seg = None
+        assert len(ak.list_symbol_table()) == 0
+
+        g = ak.GroupBy(
+            [self.make_pdarray(ak.int64, 10), self.make_pdarray(ak.str_, 10), ak.Categorical(self.make_pdarray(ak.str_, 10))]
+        )
+        assert len(ak.list_symbol_table()) > 0
+        g = None
+        assert len(ak.list_symbol_table()) == 0
+
+        d = ak.DataFrame(
+            {
+                "pda": self.make_pdarray(ak.int64, 10),
+                "s": self.make_pdarray(ak.str_, 10),
+                "cat": ak.Categorical(self.make_pdarray(ak.str_, 10)),
+                "seg": ak.SegArray(
+                    ak.arange(0, 20, 2), self.make_pdarray(ak.uint64, 20)
+                ),
```

Comment:
black suggests combining these lines (there is also one other similar instance)
`                "seg": ak.SegArray(ak.arange(0, 20, 2), self.make_pdarray(ak.uint64, 20)),
`

Reply:
updated